cli/src/commands/squash.rs

```diff
@@ -155,21 +222,106 @@ pub(crate) fn cmd_squash(
+            }
+        }
+    };
+
+    let mut tx = workspace_command.start_transaction();
+    let destination = if let Some(commit) = pre_existing_destination {
+        commit
+    } else {
+        // create the new destination commit
+        let (parent_ids, child_ids) = compute_commit_location(
+            ui,
+            tx.base_workspace_helper(),
+            args.destination.as_deref(),
+            args.insert_after.as_deref(),
+            args.insert_before.as_deref(),
+            "squash commit",
+        )?;
+        tx.base_workspace_helper()
+            .check_rewritable(sources.iter().ids().chain(child_ids.iter()))?;
+        let parent_commits: Vec<_> = parent_ids
+            .iter()
+            .map(|commit_id| {
+                tx.base_workspace_helper()
+                    .repo()
+                    .store()
+                    .get_commit(commit_id)
+            })
+            .try_collect()?;
+        let merged_tree = merge_commit_trees(tx.repo(), &parent_commits).block_on()?;
+        let mut commit_builder = tx
+            .repo_mut()
+            .new_commit(parent_ids.clone(), merged_tree.id())
+            .detach();
+        let mut description = join_message_paragraphs(&args.message_paragraphs);
+        if !description.is_empty() {
+            commit_builder.set_description(description);
+            description = add_trailers(ui, &tx, &commit_builder)?;
+        }
+        commit_builder.set_description(&description);
+        let commit = commit_builder.write(tx.repo_mut())?;
+        for child_id in &child_ids {
+            let child_commit = tx.repo().store().get_commit(child_id)?;
+            let new_parent_ids = child_commit
+                .parent_ids()
+                .iter()
+                .filter(|id| !&parent_ids.contains(id))
+                .cloned()
+                .chain(std::iter::once(commit.id().clone()))
+                .collect_vec();
+            let new_child_commit =
+                rebase_commit(tx.repo_mut(), child_commit, new_parent_ids).block_on()?;
+            if let Some(source_commit) = sources
+                .iter()
+                .position(|source| source.id() == child_id)
+                .and_then(|index| sources.get_mut(index))
+            {
+                *source_commit = new_child_commit;
+            }
+        }
+        tx.repo_mut()
+            .transform_descendants(child_ids, async |rewriter| {
+                let old_commit_id = rewriter.old_commit().id().clone();
+                let new_commit = rewriter.rebase().await?.write()?;
+                if let Some(source_commit) = sources
+                    .iter()
+                    .position(|source| source.id() == &old_commit_id)
+                    .and_then(|index| sources.get_mut(index))
+                {
+                    *source_commit = new_commit;
+                }
+                Ok(())
+            })?;
+        commit
```

Comment:
nit: we'll need to print the number of the rebased commits at the end of the operation.

Reply:
We actually need to print the number of rebased commits when there are no changes to merge into the new destination